libraries/freertos_plus/standard/freertos_plus_tcp/source/FreeRTOS_IP.c

```diff
@@ -1954,6 +1970,124 @@ uint8_t ucProtocol;
+			{
+				xLocation = 1;
+				break;
+			}
+
+			/* Parse the packet length. */
+			pxIPPacket = ipPOINTER_CAST( const IPPacket_t *, pucEthernetBuffer );
+
+			ucVersionHeaderLength = pxIPPacket->xIPHeader.ucVersionHeaderLength;
+			/* Test if the length of the IP-header is between 20 and 60 bytes,
+			and if the IP-version is 4. */
+			if( ( ucVersionHeaderLength < ipIPV4_VERSION_HEADER_LENGTH_MIN ) ||
+				( ucVersionHeaderLength > ipIPV4_VERSION_HEADER_LENGTH_MAX ) )
+			{
+				xLocation = 2;
+				break;
+			}
+			ucVersionHeaderLength = ( ucVersionHeaderLength & ( uint8_t ) 0x0FU ) << 2;
+			uxIPHeaderLength = ( UBaseType_t ) ucVersionHeaderLength;
+
+			/* Check if the complete IP-header is transferred. */
+			if( uxBufferLength < ( ipSIZE_OF_ETH_HEADER + uxIPHeaderLength ) )
+			{
+				xLocation = 3;
+				break;
+			}
+			/* Check if the complete IP-header plus protocol data have been transferred: */
+			usLength = pxIPPacket->xIPHeader.usLength;
+			usLength = FreeRTOS_ntohs( usLength );
+			if( uxBufferLength < ( size_t ) ( ipSIZE_OF_ETH_HEADER + ( size_t ) usLength ) )
+			{
+				xLocation = 4;
+				break;
+			}
+
+			/* Identify the next protocol. */
+			ucProtocol = pxIPPacket->xIPHeader.ucProtocol;
+
+			/* N.B., if this IP packet header includes Options, then the following
+			assignment results in a pointer into the protocol packet with the Ethernet
+			and IP headers incorrectly aligned. However, either way, the "third"
+			protocol (Layer 3 or 4) header will be aligned, which is the convenience
+			of this calculation. */
+			pxProtPack = ipPOINTER_CAST( ProtocolPacket_t *, &( pucEthernetBuffer[ uxIPHeaderLength - ipSIZE_OF_IPv4_HEADER ] ) );
+
+			/* Switch on the Layer 3/4 protocol. */
+			if( ucProtocol == ( uint8_t ) ipPROTOCOL_UDP )
+			{
+				/* Expect at least a complete UDP header. */
+				uxMinimumLength = uxIPHeaderLength + ipSIZE_OF_ETH_HEADER + ipSIZE_OF_UDP_HEADER;
+			}
+			else if( ucProtocol == ( uint8_t ) ipPROTOCOL_TCP )
+			{
+				uxMinimumLength = uxIPHeaderLength + ipSIZE_OF_ETH_HEADER + ipSIZE_OF_TCP_HEADER;
+			}
+			else if( ( ucProtocol == ( uint8_t ) ipPROTOCOL_ICMP ) ||
+					 ( ucProtocol == ( uint8_t ) ipPROTOCOL_IGMP ) )
+			{
+				uxMinimumLength = uxIPHeaderLength + ipSIZE_OF_ETH_HEADER + ipSIZE_OF_ICMP_HEADER;
+			}
+			else
+			{
+				/* Unhandled protocol, other than ICMP, IGMP, UDP, or TCP. */
+				xLocation = 5;
+				break;
+			}
+			if( uxBufferLength < uxMinimumLength )
+			{
+				xLocation = 6;
+				break;
+			}
+
+			uxLength = ( size_t ) usLength;
+			uxLength -= ( ( uint16_t ) uxIPHeaderLength ); /* normally minus 20 */
+
+			if( ( uxLength < ( ( size_t ) sizeof( pxProtPack->xUDPPacket.xUDPHeader ) ) ) ||
+				( uxLength > ( ( size_t ) ipconfigNETWORK_MTU - ( size_t ) uxIPHeaderLength ) ) )
+			{
+				/* Again, in a 16-bit return value there is no space to indicate an
+				error.  For incoming packets, 0x1234 will cause dropping of the packet.
```

Comment:
This comment doesn't seem to match the logic in this block.  This function returns `BaseType_t` with only `pdPASS` and `pdFAIL` values.

Reply:
Updated this comment as well